spring-webflux/src/main/java/org/springframework/web/reactive/socket/client/JettyWebSocketClient.java

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright 2002-2022 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.web.reactive.socket.client;
+
+import java.io.IOException;
+import java.net.URI;
+import java.util.concurrent.atomic.AtomicReference;
+
+import org.eclipse.jetty.client.Request;
+import org.eclipse.jetty.client.Response;
+import org.eclipse.jetty.http.HttpHeader;
+import org.eclipse.jetty.util.component.LifeCycle;
+import org.eclipse.jetty.websocket.client.ClientUpgradeRequest;
+import org.eclipse.jetty.websocket.client.JettyUpgradeListener;
+import reactor.core.publisher.Mono;
+import reactor.core.publisher.Sinks;
+
+import org.springframework.core.io.buffer.DefaultDataBufferFactory;
+import org.springframework.http.HttpHeaders;
+import org.springframework.lang.Nullable;
+import org.springframework.web.reactive.socket.HandshakeInfo;
+import org.springframework.web.reactive.socket.WebSocketHandler;
+import org.springframework.web.reactive.socket.adapter.JettyWebSocketHandlerAdapter;
+import org.springframework.web.reactive.socket.adapter.JettyWebSocketSession;
+
+public class JettyWebSocketClient implements WebSocketClient {
+
+	private final org.eclipse.jetty.websocket.client.WebSocketClient client;
+
+	public JettyWebSocketClient() {
+		this.client = new org.eclipse.jetty.websocket.client.WebSocketClient();
+		LifeCycle.start(this.client);
```

Comment:
There is a `start()` method, do you need this here?
The other constructor does not need it?
If the `start()` method is enough, please chain this constructor to the other.

Reply:
Then the constructor would need to declare `throws Exception`.

I'm not sure what do to with this, because the other clients you construct them with empty constructors and they're instantly usable, but ours needs to be started and stopped but there is no API for that on `WebSocketClient`.

Reply:
Please chain constructors.

Reply:
done